Update framework-zephyr to 2.5.0

vinceWii4 opened the pull request:
Port from Gerrit and rebase onto v2.5-branch:
- 9 first commits come from Gerrit and are rebased
- 2 new patchs:  _s2325 - Fix zephyr BMI160 I2C driver_ then _s2325 - Set anymotion interrupt on INT1_

cpavot committed:
s2005 - Added package.json fake dependency

This file is needed for platformio to build properly
though all information can actually be removed as
not used (behavior is overloaded by the checkout'ed
branch).

Change-Id: Ied5bfc188ff5ffff8bd75d1a9effc62facac41c5
Signed-off-by: Christophe PAVOT <christophe.pavot@wiifor.com>

cpavot committed:
s2005 - Added support for WBLGT01 board

Handling W4 WBLGT01 board directly in framework-zephyr
project will avoid the use of janus symbolic_link.sh
script to be able to use this board. It will be
automatically imported by platformio during framework
dependencies installation.

Change-Id: Idde9db3bf1add3489000200d9c554d2b3baec264
Signed-off-by: Christophe PAVOT <christophe.pavot@wiifor.com>

cpavot committed:
s2107 - Removed useless flag CONFIG_BOARD_WBLGT01

Change-Id: I8f5487caebbc8785fec5793b326068fe4615cf4a
Signed-off-by: Christophe PAVOT <christophe.pavot@wiifor.com>

letcheber committed:
s2153 - Add Jenkinsfile in framework-zephyr (to trigger Build in Janus)

Change-Id: I1ea54e7b50ca44c0412132fde74daba1f28f0b1d
Signed-off-by: Lisa Etcheber <lisa.etcheber@wiifor.com>

letcheber committed:
s2153 - Delete of Jenkinsfile in framework-zephyr to trigger Build in Janus : a better solution has been implemented

Change-Id: I5780e8dd121f563c23269fbec51883aa63442d1e
Signed-off-by: Lisa Etcheber <lisa.etcheber@wiifor.com>

NicoWii4 committed:
s2213 - Add bme280 in device tree

Change-Id: I3171679f9514d069c85770e6a6668481041b02d8
Signed-off-by: Nicolas <nicolas.pelissier@wiifor.com>

vinceWii4 committed:
[zephyr pull request] s2259 - Bmi160: add run time power management

This patch is an improvement of zephyr bmi160 driver.
It will be submitted to the zephyr community and might be merged in next framework-zephyr update

Change-Id: I328560c6db9bf80dfca1d214b1d66d08f9a66910
Signed-off-by: Vincent De Carvalho <vincent.decarvalho@wiifor.com>

vinceWii4 committed:
[zephyr bug fix] s2259 - Bmi160: fix activation of undersampling mode

This patch was submitted to the zehyr community and will be merged on next framework-zephyr update.

Change-Id: I61cb09559af999e4eb602d073f4287204f3c75c9
Signed-off-by: Vincent De Carvalho <vincent.decarvalho@wiifor.com>

vinceWii4 committed:
[zephyr pull request] s2290 - Fetch data in BMI160 RUNTIME power mode

This patch is an improvement of zephyr bmi160 driver.
It will be submitted to the zephyr community and might be merged in next framework-zephyr update

Change-Id: Ic0647bd5d0847c76b1381615dfd16bf1db8ab01a
Signed-off-by: Vincent De Carvalho <vincent.decarvalho@wiifor.com>

vinceWii4 committed:
s2325 - Set anymotion interrupt on INT1

Do not map interrupt at BMI160 initialization
Map the anymotion interrupt on INT1 and set latch for 2.56s when anymotion trigger is set

Change-Id: If08fa2c470e52b6b0a0c65893a09d49b2e4b8572
Signed-off-by: Vincent De Carvalho <vincent.decarvalho@wiifor.com>

vinceWii4 committed:
[zephyr bug fix] s2325 - Fix zephyr BMI160 I2C driver

This patch is a bug fix of framework-zephyr.
It will be submitted to the zephyr community and might be merged on next framework-zephyr update.

Change-Id: I51ca10fb5a47fcdb842b7a2e5f32eb2d6f1f4bbf
Signed-off-by: Vincent De Carvalho <vincent.decarvalho@wiifor.com>

ltawii reviewed (commented):
Je ne comprends pas bien d'où tu sors tous ces deltas

ltawii commented on drivers/sensor/bmi160/bmi160_trigger.c:
```diff
@@ -54,7 +54,9 @@ static void bmi160_handle_interrupts(const struct device *dev)
 	union {
 		uint8_t raw[6];
 		struct {
+#if BMI160_BUS_SPI
```
A mettre dans un patch à part

vinceWii4 replied:
Descendu dans le patch de _fix i2c driver_

vinceWii4 committed:
[zephyr bug fix] s2325 - Missing DT_DRIVER_COMPAT definition in bmi160_trigger.c

This patch is a bug fix of framework-zephyr.
It will be submitted to the zephyr community and might be merged on next framework-zephyr update.

Signed-off-by: Vincent De Carvalho <vincent.decarvalho@wiifor.com>

vinceWii4 commented:
J'ai pris en compte la remarque de Ludo, et j'ai renommé les titres des commit.
J'ai supprimé de _**[zephyr bug fix] s2325 - Fix zephyr BMI160 I2C driver**_ une mauvaise correction dont le bug visé est corrigé par le nouveau commit _**[zephyr bug fix] s2325 - Missing DT_DRIVER_COMPAT definition in bmi16…**_
